warehouse/email/ses/models.py

```diff
@@ -155,6 +155,12 @@ def _reset_transient_bounce(self):
         collector=lambda iterable: list(iterable)[-1],
     )
 
+    soft_bounced.upon(
+        deliver,
+        enter=delivered,
+        outputs=[],
```

Comment:
We probably want to reset our transient bounces in this case, because we've successfully delivered it. We can do that by adding ``_incr_transient_bounce`` to the outputs, and we'll need to set a collector too.